src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -52,22 +65,101 @@ def validate_client_parameters(namespace):
         else:
             raise ValueError("Storage account '{}' not found.".format(n.account_name))
 
-def get_content_setting_validator(settings_class):
+def validate_source_uri(namespace):
+    usage_string = 'invalid usage: supply only one of the following argument sets:' + \
+                   '\n\t   --source-uri' + \
+                   '\n\tOR --source-container --source-blob [--source-snapshot] [--source-sas]' + \
+                   '\n\tOR --source-share --source-path [--source-sas]'
+    ns = vars(namespace)
```

Comment:
I use vars so I can easily remove things from the namespace. Otherwise I have to operate on the namespace and then use "del namespace.foo" to remove things.

---
In reply to: [77393565](https://github.com/Azure/azure-cli/pull/804#discussion_r77393565) [](ancestors = 77393565)